### PR TITLE
Detect changed connections (for hightlighting fields)

### PR DESCRIPTION
**Pull request by juliasaveliff**
image

### CODE REVIEW

File: rackcity/utils/change_planner_utils.py [hunk 1]
```diff
@@ -289,13 +297,65 @@ def get_page_count_response_for_cp(request, change_plan):
...
+        # Get Asset corresponding to this live connection
+        connected_port_live = network_ports.get(
+            port_name=network_port_cp.port_name
+        )
+        if connected_port_live:
+            destination_host_live = connected_port_live.asset
+        else:
+            destination_host_live = None
+        # Check for match
+        if destination_host_in_cp != destination_host_live:
```
Comment (clairecfu) [inline rackcity/utils/change_planner_utils.py]: my brain is tired. would this miss changes in destination port names? aka connected to same hostname but a different port?

Comment (juliasaveliff) [reply rackcity/utils/change_planner_utils.py]: good catch yeah I'll update

File: rackcity/utils/change_planner_utils.py [hunk 2]
```diff
@@ -289,13 +298,69 @@ def get_page_count_response_for_cp(request, change_plan):
...
+                    real_connected_port_in_cp = NetworkPort.get(
+                        asset=related_asset,
+                        port_name=cp_connected_port_in_cp.port_name,
+                    )
+                except ObjectDoesNotExist:
+                    real_connected_port_in_cp = None
+        # Check for match
+        if connected_port_live != real_connected_port_in_cp:
+            return True
+    return False
```
Comment (juliasaveliff) [inline rackcity/utils/change_planner_utils.py]: this took some crazy brain power and drawings to get right I think it's right now it looks like it's working thank you and goodbye @clairecfu

Comment (benauriemma) [reply rackcity/utils/change_planner_utils.py]: I think you could have just checked the port name and hostname but go off queen

File: rackcity/utils/change_planner_utils.py [hunk 3]
```diff
@@ -289,13 +298,69 @@ def get_page_count_response_for_cp(request, change_plan):
...
+        # Get PDUPort corresponding to this live connection
+        pdu_port_live = power_ports.get(port_name=power_port_cp.port_name)
+        # Check for match
+        if (pdu_port_cp is None and pdu_port_live is None):
+            continue
+        if (
+            (pdu_port_cp is None and pdu_port_live)
+            or (pdu_port_cp and pdu_port_live is None)
+        ):
+            return False
```
Comment (benauriemma) [inline rackcity/utils/change_planner_utils.py]: Think this should be True!

File: rackcity/utils/change_planner_utils.py [hunk 4]
```diff
@@ -289,13 +298,69 @@ def get_page_count_response_for_cp(request, change_plan):
...
+            return True
+    return False
+
+
+def get_changes_on_asset(asset, asset_cp):
     fields = [field.name for field in Asset._meta.fields]
     changes = []
     for field in fields:
-        if getattr(asset, field) != getattr(assetCP, field):
+        if getattr(asset, field) != getattr(asset_cp, field):
```
Comment (benauriemma) [inline rackcity/utils/change_planner_utils.py]: Nice catch

File: rackcity/utils/change_planner_utils.py (hunk 2 above)
Comment (juliasaveliff) [reply rackcity/utils/change_planner_utils.py]: oh


my






god

Comment (juliasaveliff) [reply rackcity/utils/change_planner_utils.py]: except hostnames & asset numbers could have changed in change plan so i dont love that for identifying an asset actually. I stand by my long and winding path

File: rackcity/utils/change_planner_utils.py (hunk 3 above)
Comment (juliasaveliff) [reply rackcity/utils/change_planner_utils.py]: so True haah

File: rackcity/utils/change_planner_utils.py (hunk 2 above)
Comment (benauriemma) [reply rackcity/utils/change_planner_utils.py]: One could make the argument that if one disconnected host1 from host2 and connected it to host3 but then renamed host2 to host3 and renamed host3 to host2 that they actually did nothing at all

Comment (benauriemma) [reply rackcity/utils/change_planner_utils.py]: But I stand with you.

File: rackcity/utils/change_planner_utils.py (hunk 3 above)
Comment (benauriemma) [reply rackcity/utils/change_planner_utils.py]: Laughed audibly

Comment (benauriemma) [reply rackcity/utils/change_planner_utils.py]: Though I'm not proud of that

**Pull request merged**